src/analysis_and_optimization/Optimize.ml

```diff
@@ -248,23 +280,60 @@ let rec inline_function_expression propto adt fim (Expr.Fixed.{pattern; _} as e)
                 | Fun_kind.UserDefined _ -> Fun_kind.UserDefined (fname, suffix)
                 | _ -> StanLib (fname, suffix, AoS) in
               (d_list, s_list, {e with pattern= FunApp (fun_kind, es)})
-          | Some (rt, args, b) ->
-              let x = Gensym.generate ~prefix:"inline_" () in
-              let handle = handle_early_returns (Some x) in
+          | Some (rt, args, body) ->
+              let inline_return_name = gen_inline_var fname "return" in
+              let handle =
+                handle_early_returns fname (Some inline_return_name) in
               let d_list2, s_list2, (e : Expr.Typed.t) =
+                let unsized_to_sized rt =
+                  match rt with
+                  | Type.Sized _ as a -> a
+                  | Unsized aa ->
```

Comment:
This needs a lot of explanation in the code I think. Both what is doing, why it is needed, why size 0 is safe, etc

Reply:
Possibly put it in SizedType.ml (or unsizedtype) with a name like `unsafe_unsized_to_sized`

Reply:
I moved this over to `Mir_utils` as I think this is a thing we are only going to use in the optimizations. Also added a bunch of docs describing the behavior since it is an odd function